YAXLib/Caching/UdtWrapperCache.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (C) Sina Iravanian, Julian Verdurmen, axuno gGmbH and other contributors.
+// Licensed under the MIT license.
+
+#nullable enable
+
+using System;
+
+namespace YAXLib.Caching;
+
+/// <summary>
+///     Implements a singleton cache for <see cref="UdtWrapper"/>s
+///     to prevent creation of <see cref="UdtWrapper"/>s for the same type repetitively.
+/// </summary>
+internal class UdtWrapperCache : TypeCacheAbstract<UdtWrapper>
+{
+    public const int DefaultCacheSize = 200;
+
+    /// <summary>
+    ///     The singleton cache instance.
+    /// </summary>
+    private static UdtWrapperCache? _instance;
+
+    private UdtWrapperCache()
+    {
+        MaxCacheSize = DefaultCacheSize;
```

Comment:
Is 200 enough for most common use cases?

Reply:
That's the million dollar question:)

Reply:
Let's cache 500 different types, and get the million dollars :)